### ISSUE TITLE
Tests fail with Perl 5.37.10

### ISSUE THREAD

**Issue by jplesnik** [maros/MooseX-App#69]
Tests are failing since Perl 5.37.10.

If you want to see complete log please check [CPAN Testers](http://matrix.cpantesters.org/?dist=MooseX-App+1.42).

Example from log:
```
given is deprecated at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Meta/Role/Class/Base.pm line 572.
when is deprecated at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Meta/Role/Class/Base.pm line 573.
when is deprecated at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Meta/Role/Class/Base.pm line 576.
Smartmatch is deprecated at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Meta/Role/Class/Base.pm line 741.

#   Failed test 'no warnings'
#   at /usr/share/perl5/Test/Builder.pm line 193.
# There were 24 warning(s)
#     Previous test 0 ''
#     given is deprecated at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Utils.pm line 237.
#  at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Utils.pm line 237.
# 	require MooseX/App/Utils.pm called at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Exporter.pm line 11
# 	MooseX::App::Exporter::BEGIN() called at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Exporter.pm line 11
# 	eval {...} called at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App/Exporter.pm line 11
# 	require MooseX/App/Exporter.pm called at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App.pm line 14
# 	MooseX::App::BEGIN() called at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App.pm line 14
# 	eval {...} called at /builddir/build/BUILD/MooseX-App-1.42/blib/lib/MooseX/App.pm line 14
# 	require MooseX/App.pm called at t/testlib/Test01.pm line 4
# 	Test01::BEGIN() called at t/testlib/Test01.pm line 4
# 	eval {...} called at t/testlib/Test01.pm line 4
# 	require Test01.pm called at t/01_basic.t line 11
# 	main::BEGIN() called at t/01_basic.t line 11
# 	eval {...} called at t/01_basic.t line 11
# 
# ----------
```

**Comment by maros** [comment]
Merged the PR, thank you for the contribution

**Comment by maros** [comment]
Released on cpan as 1.43
